src/snmalloc/ds_core/stats.h

```diff
@@ -0,0 +1,68 @@
+#include "defines.h"
+
+#include <atomic>
+#include <cstddef>
+
+namespace snmalloc
+{
+  /**
+   * Very basic statistic that tracks current and peak values.
+   */
+  class Stat
+  {
+  private:
+    std::atomic<size_t> curr{0};
+    std::atomic<size_t> peak{0};
+
+  public:
+    void increase(size_t amount)
+    {
+      size_t c = (curr += amount);
+      size_t p = peak.load(std::memory_order_relaxed);
+      while (c > p)
```

Comment:
Huh.  I'm a little surprised that `std::atomic` doesn't have `min`/`max` ops.  I wonder if this loop lowers to RISC-V's `AMOMAX`.